src/command/global/settings.ts

```diff
@@ -0,0 +1,163 @@
+                .setCustomId(Toggles.QUEUE)
+                .setLabel("Empty Queue")
+                .setEmoji("🧠")
+                .setStyle(await getStyle(discordId, Toggles.QUEUE))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.REIMB)
+                .setLabel("Reimb Ready")
+                .setEmoji("💰")
+                .setStyle(await getStyle(discordId, Toggles.REIMB))
+        );
+
+    const row2 = new MessageActionRow()
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.EVENT)
+                .setLabel("New Event")
+                .setEmoji("❗")
+                .setStyle(await getStyle(discordId, Toggles.EVENT))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.MAIL)
+                .setLabel("New Mail")
+                .setEmoji("📧")
+                .setStyle(await getStyle(discordId, Toggles.MAIL))
+        )
+        .addComponents(
+            new MessageButton()
+                .setCustomId(Toggles.PAUSED)
+                .setLabel("Pause Notifications")
+                .setEmoji("💤")
+                .setStyle(await getStyle(discordId, Toggles.PAUSED))
+        );
+
+    return [row, row2];
+}
+
+function getCurrentSetting(setting: string, settings: NotificationSettings) {
+    if (setting == Toggles.WAR) {
+        return settings.war_flag;
+    }
+    if (setting == Toggles.QUEUE) {
+        return settings.queue_flag;
+    }
+    if (setting == Toggles.REIMB) {
+        return settings.reimb_flag;
+    }
+    if (setting == Toggles.ENEMY) {
+        return settings.enemy_flag;
+    }
+    if (setting == Toggles.MAIL) {
+        return settings.mail_flag;
+    }
+    if (setting == Toggles.EVENT) {
+        return settings.event_flag;
+    }
+    if (setting == Toggles.PAUSED) {
+        return settings.paused_flag;
+    }
+}
+
+export const handler = async (interaction: CommandInteraction) => {
+    const settingsRepository = getCustomRepository(
+        NotificationSettingsRepository
+    );
+    const response = await settingsRepository.getUserByDiscordId(
+        interaction.user.id
+    );
+    if (response && !response.valid_key) {
+        interaction.reply({
+            content:
+                "Please save your API with the bot in order to use this feature",
+        });
+    } else {
+        await interaction.reply({
+            content: "Not implemented",
+            components: await createRows(interaction.user.id),
+        });
+    }
+};
+
+export async function handleButton(interaction: ButtonInteraction) {
+    const settingsRepository = getCustomRepository(
+        NotificationSettingsRepository
+    );
+    const values = await settingsRepository.getUserByDiscordId(
+        interaction.user.id
+    );
+    if (values) {
+        const existingSetting = getCurrentSetting(interaction.customId, values);
```

Comment:
minor point, but this could be condensed to a single query

Reply:
what do you mean?